src/createContext.lua

```diff
@@ -0,0 +1,92 @@
+local Symbol = require(script.Parent.Symbol)
+local Binding = require(script.Parent.Binding)
+local oneChild = require(script.Parent.oneChild)
+local Children = require(script.Parent.PropMarkers.Children)
+local PureComponent = require(script.Parent.PureComponent)
+
+local function createProvider(context)
+	local Provider = PureComponent:extend("Provider")
+
+	function Provider:init(props)
+		self.binding, self.updateValue = Binding.create(props.value)
+
+		local key = context.key
+		self._context[key] = self.binding
+	end
+
+	function Provider:didUpdate()
+		self.updateValue(self.props.value)
```

Comment:
Let's check that `prevProps.value ~= self.props.value` before firing this method. This will let us avoid firing updates when we don't need to.